CRM/Contact/BAO/Contact/Permission.php

```diff
@@ -33,6 +33,96 @@
 class CRM_Contact_BAO_Contact_Permission {
 
   /**
+   * Check which of the given contact IDs the logged in user
+   *   has permissions for the operation type according
+   *
+   * Caution: general permissions (like 'edit all contacts')
+   *
+   * @param array $contact_ids
+   *   Contact IDs.
+   * @param int|string $type the type of operation (view|edit)
+   *
+   * @see CRM_Contact_BAO_Contact_Permission::allow
+   *
+   * @return array
+   *    list of contact IDs the logged in user has the given permission for
+   */
+  public static function allowList($contact_ids, $type = CRM_Core_Permission::VIEW) {
+    $result_set = array();
+    if (empty($contact_ids)) {
+      // empty contact lists would cause trouble in the SQL. And be pointless.
+      return $result_set;
+    }
+    $contact_id_list = implode(',', $contact_ids);
```

Comment:
trivial point - but this might be better after the 'early return' for 'edit all contacts' etc